src/api_server/src/request/entropy.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use vmm::rpc_interface::VmmAction;
+use vmm::vmm_config::entropy::EntropyDeviceConfig;
+
+use crate::parsed_request::{Error, ParsedRequest};
+use crate::request::Body;
+
+pub(crate) fn parse_put_entropy(body: &Body) -> Result<ParsedRequest, Error> {
+    let cfg = serde_json::from_slice::<EntropyDeviceConfig>(body.raw())?;
+    Ok(ParsedRequest::new_sync(VmmAction::SetEntropyDevice(cfg)))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_put_entropy_request() {
+        assert!(parse_put_entropy(&Body::new("invalid_payload")).is_err());
```

Comment:
Nit it is usually better to match the specific value e.g. `assert_eq!(..., Err(...))`.

Reply:
same as before, I don't think the added value justifies the complexity.